src/group_theory/schur_zassenhaus.lean

```diff
@@ -162,21 +164,204 @@ begin
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step5 {P : sylow (fintype.card N).min_fac N} : P.1 ≠ ⊥ :=
+begin
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  exact P.ne_bot_of_dvd_card (fintype.card N).min_fac_dvd,
+end
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step6 : is_p_group (fintype.card N).min_fac N :=
+begin
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  refine sylow.nonempty.elim (λ P, P.2.of_surjective P.1.subtype _),
+  rw [←monoid_hom.range_top_iff_surjective, subtype_range],
+  haveI : (P.1.map N.subtype).normal := normalizer_eq_top.mp
+    (step1 h1 h2 h3 (P.1.map N.subtype).normalizer P.normalizer_sup_eq_top),
+  exact (step3 h1 h2 h3 P.1).resolve_left (step5 h1 h2 h3),
+end
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step7 : is_commutative N :=
+begin
+  haveI := N.bot_or_nontrivial.resolve_left (step0 h1 h2 h3),
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  exact ⟨⟨λ g h, eq_top_iff.mp ((step3 h1 h2 h3 N.center).resolve_left
+    (step6 h1 h2 h3).bot_lt_center.ne') (mem_top h) g⟩⟩,
+end
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+lemma contradiction : false :=
+begin
+  haveI := step7 h1 h2 h3,
+  exact not_exists_of_forall_not h3 (exists_right_complement'_of_coprime_aux h1),
+end
+
+end schur_zassenhaus_induction
+
+variables {n : ℕ} {G : Type u} [group G]
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma exists_right_complement'_of_coprime_aux' [fintype G] (hG : fintype.card G = n)
+  {N : subgroup G} [N.normal] (hN : nat.coprime (fintype.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+begin
+  unfreezingI { revert G },
+  apply nat.strong_induction_on n,
+  rintros n ih G _ _ rfl N _ hN,
+  exactI not_forall_not.mp
+    (schur_zassenhaus_induction.contradiction hN (λ G' _ _ hG', by { apply ih _ hG', refl })),
+end
+
+/-- **Schur-Zassenhaus** for normal subgroups:
+  If `H : subgroup G` is normal, and has order coprime to its index, then there exists a
+  subgroup `K` which is a (right) complement of `H`. -/
+theorem exists_right_complement'_of_coprime [fintype G]
+  {N : subgroup G} [N.normal] (hN : nat.coprime (fintype.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+exists_right_complement'_of_coprime_aux' rfl hN
+
+/-- **Schur-Zassenhaus** for normal subgroups:
+  If `H : subgroup G` is normal, and has order coprime to its index, then there exists a
+  subgroup `K` which is a (right) complement of `H`. -/
+theorem exists_right_complement'_of_coprime'
+  {N : subgroup G} [N.normal] (hN : nat.coprime (nat.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+begin
```

Comment:
I think it would be good to include `of_fintype` in the first name, and remove the `'` in the second name. That way it is directly clear where the difference lies. (Of course the types are also a bit different, but that's fine.)